bip-0322.mediawiki

```diff
@@ -13,119 +13,99 @@
 
 == Abstract ==
 
-A standard for interoperable generic signed messages based on the Bitcoin Script format.
-
-== Background ==
-
-* Assume two actors, a prover <code>P</code> and a verifier <code>V</code>.
-* <code>P</code> wants to prove that they own the private key <code>k</code> associated with a given address <code>A</code> (which in turn is derived from the pubkey <code>kG</code>).
-* Let <code>V</code> generate a message <code>M</code> and hand this to <code>P</code>.
-* <code>P</code> generates a signature <code>S</code> by signing the message <code>M</code> using <code>k</code>. Given <code>S</code>, <code>V</code> can prove that <code>P</code> has the private key associated with <code>A</code>.
-
-The astute reader will notice that the above is missing a critical part, namely the pubkey <code>kG</code>, without which the verifier cannot actually verify the message. The current message signing standard solves this via a cryptographic trick, wherein the signature <code>S</code> above is a special "recoverable signature" type. Given the message <code>M</code> and the signature <code>S</code>, it is then possible to recover the pubkey <code>kG</code>. The system thus derives the address for the pubkey <code>kG</code>, and if it does not match <code>A</code>, the proof is deemed invalid.
-
-While this is a neat trick, it unnecessarily restricts and complicates the message signing mechanism; for instance, it is currently not possible to sign a message for a P2SH address, because there is no pubkey to recover from the resulting signature.
+A standard for interoperable signed messages based on the Bitcoin Script format, either for proving fund availability, or committing to a message as a recipient of an invoice address.
 
 == Motivation ==
 
-The current message signing standard only works for P2PKH (1...) addresses. By extending it to use a Bitcoin Script based approach, it could be made more generic without causing a too big burden on implementers, who most likely have access to Bitcoin Script interpreters already.
-
-== Specification ==
+The current message signing standard only works for P2PKH (1...) invoice addresses. By extending it to use a Bitcoin Script based approach, it could be made more generic without causing a too big burden on implementers, who most likely have access to Bitcoin Script interpreters already.
 
-A new structure <code>SignatureProof</code> is added, which is a simple serializable scriptSig & witness container.
+Additionally, the current message signing only proves that the message has been committed to by the recipient of a given invoice address.
+It does not prove anything about the invoice address itself, nor that the signer has access to the private keys used to implement this invoice.
```

Comment:
I can forward the message to Fred, who owns the keys, and he signs it and hands it back to me. Now I don't have the keys myself, but I am able to produce that signature for you, because Fred allows me. He may not allow me to send from a UTXO associated with that invoice address though, but that's another story.